### PR TITLE
Handle paths with spaces when editing credentials

### LINKED ISSUE #41617: change_credentials_in_system_editor breaks with spaces in Windows user name
### Steps to reproduce

Run `rails credentials:edit` while using a windows username that contains spaces. In my case, it's "Lucius Riccio" (since this was my dad's laptop a decade ago)

### Expected behavior

Open a temporary file with a path like `C:/Users/Lucius Riccio/AppData/Local/Temp/22716.credentials.yml`
### Actual behavior

Opens two temporary files with vscode, like `Lucius` and a file named  Riccio/AppData/Local/Temp/22716.credentials.yml.

### System configuration
**Rails version**:
6.1.2.1
**Ruby version**:
3.0.0

[`change_credentials_in_system_editor`](https://github.com/rails/rails/blob/fdba9304c2652af674fdb9cf3704ff5a64af92d7/railties/lib/rails/commands/credentials/credentials_command.rb#L92) should handle paths with spaces.

### PR DESCRIPTION
**Pull request by ghiculescu**
Fixes https://github.com/rails/rails/issues/41617

### CODE REVIEW

**Review by Peter-Lavigne — APPROVED** [review]

**Comment by dorianmariefr** [comment]
Should you use `Shellwords.escape`?

File: railties/lib/rails/commands/credentials/credentials_command.rb [hunk 1]
```diff
@@ -91,7 +91,7 @@ def ensure_credentials_have_been_added
 
         def change_credentials_in_system_editor
           credentials.change do |tmp_path|
-            system("#{ENV["EDITOR"]} #{tmp_path}")
+            system("#{ENV["EDITOR"]} \'#{tmp_path}\'")
```
Comment (ghiculescu) [inline railties/lib/rails/commands/credentials/credentials_command.rb:94]:
```suggestion
            system("#{ENV["EDITOR"]} #{Shellwords.escape(tmp_path)}")
```

File: railties/lib/rails/commands/credentials/credentials_command.rb [hunk 2]
```diff
@@ -91,7 +92,7 @@ def ensure_credentials_have_been_added
 
         def change_credentials_in_system_editor
           credentials.change do |tmp_path|
-            system("#{ENV["EDITOR"]} #{tmp_path}")
+            system("#{ENV["EDITOR"]} #{Shellwords.escape(tmp_path)}")
```
Comment (jonathanhefner) [inline railties/lib/rails/commands/credentials/credentials_command.rb:95]: Apologies for dogpiling :dog:, but Ruby provides a [`system`](https://docs.ruby-lang.org/en/2.7.0/Kernel.html#method-i-system) overload that can be helpful here:

```suggestion
            system(ENV["EDITOR"], tmp_path)
```

Comment (ghiculescu) [reply railties/lib/rails/commands/credentials/credentials_command.rb:95]: Oh taht's cooool

File: railties/lib/rails/commands/credentials/credentials_command.rb [hunk 3]
```diff
@@ -91,7 +91,7 @@ def ensure_credentials_have_been_added
 
         def change_credentials_in_system_editor
           credentials.change do |tmp_path|
-            system("#{ENV["EDITOR"]} #{tmp_path}")
+            system(ENV["EDITOR"], tmp_path)
```
Comment (jonathanhefner) [inline railties/lib/rails/commands/credentials/credentials_command.rb:94]: Ah, I didn't realize `tmp_path` is a `Pathname`.  Adding a `to_s` should fix the build:

```suggestion
            system(ENV["EDITOR"], tmp_path.to_s)
```

**Comment by jonathanhefner** [comment]
The build is now failing due to this test: https://github.com/rails/rails/blob/317547e0e7fb8051ed92aa8f705bde5646fe27d8/railties/test/commands/credentials_test.rb#L39-L45

Specifically, the first editor command is a partial command string, not an executable.  I didn't think of it before, but we do need to support `ENV["EDITOR"]` values that include arguments (e.g. `"subl -w"`).  We could use `Shellwords.split` for that (i.e. `system(*Shellwords.split(ENV["EDITOR"]), tmp_path.to_s)`), but that won't work with the `>` in the test command.  At this point, we're probably better off just using `Shellwords.escape` like you had before.  Sorry for the noise. :facepalm:

**Comment by ghiculescu** [comment]
Ugh, I should actually run the tests before committing. Will fix it properly now.

**Comment by ghiculescu** [comment]
> At this point, we're probably better off just using `Shellwords.escape` like you had before. Sorry for the noise.

All good! I agree - have switched back to that and confirms test pass 👍

**Review by jonathanhefner — APPROVED** [review]

File: railties/lib/rails/commands/credentials/credentials_command.rb (hunk 2 above)
Comment (casperisfine) [inline railties/lib/rails/commands/credentials/credentials_command.rb:95]: Have you considered `system(ENV["EDITOR"], tmp_path)`, that would perfectly take care of the escaping and avoid a subshell.

Comment (ghiculescu) [reply railties/lib/rails/commands/credentials/credentials_command.rb:95]: I did, but then this test failed

https://github.com/rails/rails/blob/d364cfb34ea5d57f314e6e1449a8ca63029623a7/railties/test/commands/credentials_test.rb#L41

https://buildkite.com/rails/rails/builds/79105#5b1fa71b-14c0-4480-a298-afe249b6f6e2

Comment (casperisfine) [reply railties/lib/rails/commands/credentials/credentials_command.rb:95]: Ah!

**Pull request merged**